src/libmeasurement_kit/ooni/facebook_messenger.cpp

```diff
@@ -0,0 +1,220 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include "private/common/fcompose.hpp"
+#include "private/common/utils.hpp"
+#include "private/ooni/utils.hpp"
+#include "private/ooni/constants.hpp"
+#include <measurement_kit/ooni.hpp>
+
+namespace mk {
+namespace ooni {
+
+using namespace mk::report;
+
+static const std::map<std::string, std::string>
+    &FB_SERVICE_HOSTNAMES = {
```

Comment:
I seem to recall that the most robust C++ way to initialize global objects is through factory functions. I will see whether I can find a link regarding this. Going on memory, not doing that led to strange behavior sometimes.

I also found it strange for the variable to be a reference type. But probably it is initialized as a reference to an anonymous object initialized by the initialiser list?